src/Filter/UnitRefundFilter.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Filter;
+
+use Sylius\RefundPlugin\Model\UnitRefundInterface;
+
+class UnitRefundFilter implements UnitRefundFilterInterface
+{
+    public function filterUnitRefunds(array $units, string $unitRefundClass): array
+    {
+        return array_values(
+            array_filter($units, function ($units) use ($unitRefundClass): bool {
+                return $units instanceof UnitRefundInterface && $units instanceof $unitRefundClass;
```

Comment:
Do we even expect `units` to be something different then UnitRefundInterface? What would you say, that we finalize either a custom collection class or just trigger an exception if `$units instanceof UnitRefundInterface` is not true?